driver-core/src/main/java/com/datastax/driver/core/Connection.java

```diff
@@ -519,12 +523,10 @@ public String toString() {
 
     public static class Factory {
 
-        private final ExecutorService bossExecutor = Executors.newCachedThreadPool();
-        private final ExecutorService workerExecutor = Executors.newCachedThreadPool();
         public final HashedWheelTimer timer = new HashedWheelTimer(new ThreadFactoryBuilder().setNameFormat("Timeouter-%d").build());
 
-        private final ChannelFactory channelFactory = new NioClientSocketChannelFactory(bossExecutor, workerExecutor);
-        private final ChannelGroup allChannels = new DefaultChannelGroup();
+        private final EventLoopGroup eventLoopGroup = new NioEventLoopGroup();
+        private final ChannelGroup allChannels = new DefaultChannelGroup(GlobalEventExecutor.INSTANCE);
```

Comment:
At a cursory glance, this executor is only used to execute listeners registered on the `ChannelGroupFuture`s returned by `ChannelGroup` operations.
We only ever call one of these operations, namely `allChannels.close()` in `Factory#shutdown`, and we don't register any listener on it.
And even if we did, there's only one factory per cluster, one channel group per factory, and this happens only at cluster shutdown. So I think we're safe regarding concurrency on this executor.